code/datums/status_effects/status_effect.dm

```diff
@@ -17,7 +17,7 @@
 		owner = new_owner
 	if(owner)
 		LAZYADD(owner.status_effects, src)
-	addtimer(CALLBACK(src, .proc/start_ticking), 1) //Give us time to set any variables
+	addtimer(CALLBACK(src, .proc/start_ticking), 0) //Give us the minimum possible time to set any variables
```

Comment:
Plus looking over the code the his grace status effect doesn't care about which his grace you're holding, only that you're holding one.